semgrep/semgrep/core_runner.py

```diff
@@ -285,113 +348,89 @@ def _run_rules_direct_to_semgrep_core(
-                            stderr = None
-
-                        core_run = sub_run(
-                            cmd,
-                            stdout=subprocess.PIPE,
-                            stderr=stderr,
-                            preexec_fn=setrlimits_preexec_fn,
-                        )
-                        output_json = self._extract_core_output(rule, core_run)
-                        core_output = CoreOutput.parse(output_json, RuleId(rule.id))
-
-                        if "time" in output_json:
-                            self._add_match_times(
-                                rule, profiling_data, output_json["time"]
-                            )
-
-                    # end with tempfile.NamedTemporaryFile(...) ...
-                    outputs[rule].extend(core_output.rule_matches(rule))
-                    parsed_errors = [e.to_semgrep_error() for e in core_output.errors]
-                    for err in core_output.errors:
-                        if err.is_timeout():
-                            assert err.path is not None
-
-                            file_timeouts[err.path] += 1
-                            if (
-                                self._timeout_threshold != 0
-                                and file_timeouts[err.path] >= self._timeout_threshold
-                            ):
-                                max_timeout_files.add(err.path)
-                    errors.extend(parsed_errors)
-            # end for language ...
-        # end for rule ...
+            with tempfile.NamedTemporaryFile(
+                "w", suffix=".yaml"
+            ) as rule_file, tempfile.NamedTemporaryFile("w") as target_file:
+                targets_with_rules = self._get_targets(
+                    rules, target_manager, all_targets
+                )
+                target_file.write(json.dumps(targets_with_rules))
+                target_file.flush()
+
+                yaml = YAML()
+                yaml.dump({"rules": [rule._raw for rule in rules]}, rule_file)
+                rule_file.flush()
+
+                # Run semgrep
+                cmd = [SemgrepCore.path()] + [
+                    "-json",
+                    "-rules",
+                    rule_file.name,
+                    "-j",
+                    str(self._jobs),
+                    "-targets",
+                    target_file.name,
+                    "-use_parsing_cache",
+                    semgrep_core_ast_cache_dir,
+                    "-timeout",
+                    str(self._timeout),
+                    "-max_memory",
+                    str(self._max_memory),
+                    "-json_time",
+                ]
+
+                if self._optimizations != "none":
+                    cmd.append("-fast")
+
+                stderr: Optional[int] = subprocess.PIPE
+                if is_debug():
+                    cmd += ["-debug"]
+                    stderr = None
+
+                core_run = sub_run(
```

Comment:
To prevent code duplication ( and make core command dump actually accurate to what is being run )probably better to print it here instead of duplicating work

Reply:
saaaaaaad but fair